pkg/internal/compare/image.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compare
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+// Images compares the given images to each other and returns an error if they
+// differ.
+func Images(imgs ...v1.Image) error {
```

Comment:
This seems like it would be a lot simpler to implement if it only accepted two images, and probably a lot easier to understand resulting differences. What's the use case for wanting to compare 3+ images at once?

If you need to compare 3+ images at once, at least add tests to cover the 3+ case.

Reply:
I think it's the same amount of code to compare 2 or N, so I think implementing the more flexible thing makes sense.

Updated the tests to cover 3+ -- we can revisit this decision if we ever want to move out of pkg/internal/

Reply:
Can you document that errors will be phrased in terms of comparisons of image N vs N-1?

This means that `compare.Images(a, a, b, a)` will tell you `images[2] != images[1]` and `images[3] != images[2]` but won't make it very clear that images 0, 1, 3 were all the same.

The code for this is roughly the same, but the semantics of how you phrase differences can get very complex (to write, and to read) pretty quickly.

Reply:
Dropped down to just 2 things.